src/analysis/calculus/cont_diff.lean

```diff
@@ -2656,6 +2830,11 @@ begin
   exact iterated_fderiv_within_add_apply hf hg unique_diff_on_univ (set.mem_univ _),
 end
 
+lemma iterated_fderiv_add_apply' {i : ℕ} {f g : E → F} (hf : cont_diff 𝕜 i f)
+  (hg : cont_diff 𝕜 i g) :
+  iterated_fderiv 𝕜 i (λ x, f x + g x) x = iterated_fderiv 𝕜 i f x + iterated_fderiv 𝕜 i g x :=
+iterated_fderiv_add_apply hf hg
```

Comment:
What is the reason for having this lemma?

Reply:
When you see `iterated_fderiv 𝕜 i (λ x, f x + g x) x` you can not use `rw iterated_fderiv_add_apply` as this is syntactically different. While it works with the primed version.

Reply:
My point is that we don't have two lemmas for `deriv` or the differentiable statements (or for `sub`, `neg`, etc). Looking at these it seems that the `λ` version might actually be the better spelling.

Reply:
I think the current `fderiv` file predates the time where one could write transparently `f + g`, so the fact that it does not use this syntax is not really a useful data point. I agree that it would be better to settle on one of the two formulations, but it's not clear to me which one is better (`f + g` is way more readable, but not always what we get syntactically). So I went for a new lemma in this PR for minimal disruption. I can remove it from this PR if you prefer, but I'd rather not do a refactor to change all of them in mathlib in one or the other form: it's probably better for after the port.

Reply:
I would be happy if you add a note that this lemma is slightly redundant, but the refactor is a Lean 4 thing.